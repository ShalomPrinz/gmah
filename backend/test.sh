#!/bin/bash

if [[ $1 == "--coverage" ]]; then
    coverage run -m unittest discover tests
    coverage html
else
    python -m unittest discover tests
fi
