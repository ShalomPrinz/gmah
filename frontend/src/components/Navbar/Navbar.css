.navbar-brand {
  color: white !important;
  margin-left: 1rem;
}

.navbar {
  align-items: center;
  display: flex;
  font-size: 1.2rem;
  justify-content: center;
  position: sticky;
  top: 0;
  vertical-align: middle;
}

.nav-links {
  padding: 0.5rem 1rem;
  text-decoration: none;
}

.nav-item {
  line-height: 40px;
  margin-right: 1.6rem;
}

.nav-dropdown {
  background: inherit;
  font-weight: lighter;
}

.nav-dropdown:hover:after {
  background: initial;
}

.nav-dropdown-item {
  border: 0 !important;
  color: black !important;
  display: flex;
  justify-content: center;
  text-decoration: none;
}

.nav-dropdown-item:hover {
  background-color: moccasin !important;
  font-weight: bold;
}

.dropdown-toggle {
  color: white !important;
}

.dropdown-toggle.nav-link::after {
  margin-right: .6rem;
}

.offcanvas {
  background-color: var(--default-soft-bg-color) !important;
  color: var(--default-color) !important;
}
