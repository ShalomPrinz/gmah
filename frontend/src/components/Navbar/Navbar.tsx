import { useState } from "react";

import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import { NavLink } from "react-router-dom";
import ConditionalList from "../ConditionalList";
import IconComponent from "../Icon";

import "./Navbar.css";

interface Page {
  id: number;
  name: string;
  url: string;
}

const pageCallback = ({ name, url }: Page) => (
  <NavLink className="nav-item nav-links fw-light" to={url}>
    {name}
  </NavLink>
);

interface NavbarProps {
  pages: Page[];
}

function AppNavbar({ pages }: NavbarProps) {
  const [expanded, setExpanded] = useState(false);

  const icon = expanded ? "navbarExpanded" : "navbarClosed";
  const setNotExpanded = () => expanded && setExpanded(!expanded);

  return (
    <Navbar
      className="navbar bg-default"
      expand="lg"
      expanded={expanded}
      style={{ direction: "rtl" }}
    >
      <NavLink onClick={setNotExpanded} className="navbar-brand fs-1" to="/">
        <img
          height="90px"
          className="p-2 m-3"
          src="/src/res/logo.png"
          title={'גמ"ח אבישי'}
        />
        {'גמ"ח אבישי'}
      </NavLink>
      <Navbar.Toggle
        aria-controls="responsive-navbar-nav"
        onClick={() => setExpanded(!expanded)}
      >
        <IconComponent icon={icon} />
      </Navbar.Toggle>
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav onClick={setNotExpanded}>
          <ConditionalList itemCallback={pageCallback} list={pages} />
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
}

function NavbarWrapper() {
  const pages = [
    {
      id: 0,
      name: "כל המידע",
      url: "/families",
    },
    {
      id: 1,
      name: "ניהול",
      url: "/manage",
    },
  ];

  return <AppNavbar pages={pages} />;
}

export default NavbarWrapper;
