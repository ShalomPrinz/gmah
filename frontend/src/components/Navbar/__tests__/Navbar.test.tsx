import { describe, expect, it } from "vitest";

import { renderWithRouter } from "../../../../test";

import Navbar from "../Navbar";

describe("Navbar", () => {
  describe("snapshot", () => {
    it("should render navbar", () => {
      const { asFragment } = renderWithRouter(<Navbar />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
