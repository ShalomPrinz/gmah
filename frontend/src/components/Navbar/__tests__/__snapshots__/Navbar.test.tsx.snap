// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > snapshot > should render navbar 1`] = `
<DocumentFragment>
  <nav
    class="navbar bg-default navbar navbar-expand-lg navbar-light"
    style="direction: rtl;"
  >
    <a
      aria-current="page"
      class="navbar-brand fs-1 active"
      href="/"
    >
      <img
        class="p-2 m-3"
        height="90px"
        src="/src/res/logo.png"
        title="גמ\\"ח אבישי"
      />
      גמ"ח אבישי
    </a>
    <div
      class="navbar-collapse collapse"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-links fw-light"
          href="/families"
        >
          כל המידע
        </a>
        <a
          class="nav-item nav-links fw-light"
          href="/manage"
        >
          ניהול
        </a>
        <a
          class="nav-item nav-links fw-light"
          href="/managers"
        >
          אחראי נהגים
        </a>
        <a
          class="nav-item nav-links fw-light"
          href="/reports"
        >
          דוחות קבלה
        </a>
        <a
          class="nav-item nav-links fw-light"
          href="/month"
        >
          מעקב חודשי
        </a>
      </div>
    </div>
    <button
      class="fs-1 mx-5 btn btn-transparent"
      title="חזור אחורה"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-share fa-flip-horizontal "
        color="white"
        data-icon="share"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M307 34.8c-11.5 5.1-19 16.6-19 29.2v64H176C78.8 128 0 206.8 0 304C0 417.3 81.5 467.9 100.2 478.1c2.5 1.4 5.3 1.9 8.1 1.9c10.9 0 19.7-8.9 19.7-19.7c0-7.5-4.3-14.4-9.8-19.5C108.8 431.9 96 414.4 96 384c0-53 43-96 96-96h96v64c0 12.6 7.4 24.1 19 29.2s25 3 34.4-5.4l160-144c6.7-6.1 10.6-14.7 10.6-23.8s-3.8-17.7-10.6-23.8l-160-144c-9.4-8.5-22.9-10.6-34.4-5.4z"
          fill="currentColor"
        />
      </svg>
    </button>
  </nav>
</DocumentFragment>
`;
